scripts/pre-commit

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -e
+
+echo "Running Pre-Commit"
+
+FILES=`git diff --staged --name-only | grep '.*\.go' | cat`
+
+[ -z "$FILES" ] && exit 0
```

Comment:
Actually, there might be a major problem with this line. If `[ -z "$FILES" ]` returns 1 (ie. there are files), then `set -e` causes the program to exit immediately. I think you can fix this by changing the line to:

```
[ -z "$FILES" ] && echo "Pre-commit Vacuously Passed" && exit 0 || true
```

But, double check both branches to makes sure.